src/SFA.DAS.ProviderRegistrations/Application/Commands/SendInvitationEmailCommand/SendInvitationEmailCommandHandler.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using NServiceBus;
+using SFA.DAS.Notifications.Messages.Commands;
+using SFA.DAS.ProviderRegistrations.Application.Queries.GetProviderByUkprnQuery;
+using SFA.DAS.ProviderRegistrations.Configuration;
+
+namespace SFA.DAS.ProviderRegistrations.Application.Commands.SendInvitationEmailCommand
+{
+    public class SendInvitationEmailCommandHandler : AsyncRequestHandler<SendInvitationEmailCommand>
+    {
+        private const string NotificationTemplateId = "ProviderInviteEmployerNotification";
+        private readonly IMessageSession _publisher;
+        private readonly IMediator _mediator;
+        private readonly ProviderRegistrationsSettings _configuration;
+
+        public SendInvitationEmailCommandHandler(IMediator mediator, IMessageSession publisher, ProviderRegistrationsSettings configuration)
+        {
+            _publisher = publisher;
+            _mediator = mediator;
+            _configuration = configuration;
+        }
+
+        protected override async Task Handle(SendInvitationEmailCommand request, CancellationToken cancellationToken)
+        {
+            var provider = await _mediator.Send(new GetProviderByUkprnQuery(request.Ukprn), cancellationToken);
+
+            var tokens = new Dictionary<string, string>()
+            {
+                { "provider_organisation", provider.ProviderName },
+                { "provider_name", request.ProviderFullName },
+                { "employer_organisation", request.EmployerOrganisation },
+                { "employer_name", request.EmployerFullName },
+                { "invitation_link", $"{_configuration.EmployerAccountsBaseUrl}/service/register/{request.CorrelationId}" },
+                { "unsubscribe_training_provider", $"{_configuration.EmployerAccountsBaseUrl}/service/unsubscribe/{request.CorrelationId}" }
+            };
+
+            // Reply to and Subject switched out by template.
+            await _publisher.Send(new SendEmailCommand(NotificationTemplateId, request.EmployerEmail, "noreply@sfa.gov.uk", tokens, "Invitation to Apprenticeship Service"));
```

Comment:
Should these be config values?

Reply:
The reply to address and the subject all get switched out by the template that is stored in the notify portal, i looked at other places in code and some just send "x" for both as Notifications previously had validation that rejected a message if this wasn't set. ( I think its for when people decide to use send grid to test an email before dropping it into a template ) As a result i chose to at least put in some values that seemed to make sense but if it causes more confusion than it solves i can revert to just "x" like others.